controllers/ControllerAdmin.php

```diff
@@ -3,201 +3,212 @@
+use OC_Blog\Tools\ControllerFactory;
 use OC_Blog\Tools\Session;
 
 
-class ControllerAdmin {
-
-	private array $_params;
-	private array $_method;
-	private object $_twig;
-	private object $_userManager;
-	private object $_adminManager;
-	private object $_postsManager;
-	private object $_commentManager;
-	private string $_server;
-
-	public function __construct($method, $twig, $params){
-		$this->_method = $method;
-		$this->_twig = $twig;
-		$this->_params = $params;
-		$this->_userManager = new AuthManager();
-		$this->_adminManager = new AdminManager();
-		$this->_postsManager = new PostsManager();
-		$this->_commentManager = new CommentsManager();
-		$this->_server = ( new ConstantGlobal(ServerRequest::fromGlobals()) )->getServerName()['SERVER_NAME'];
-
-		$target = $method[2];
-		if (method_exists(ControllerAdmin::class, $target) ) {
-			$this->$target();
-		}else{
-			echo $this->_twig->render('404.twig');
-		}
-
-	}
+class ControllerAdmin extends ControllerFactory {
 
+	/**
+	 * Affiche la page profil de l'utilisateur.
+	 */
 	public function profile(){
 
 		$key = (new Session)->getKey('user');
-		if (isset($this->_params['identifiant'])){
+
+		if (isset($this->getPost()['identifiant'])){
 			$this->checkMailProfile();
-		}elseif (isset($this->_params['password'])){
+		}elseif (isset($this->getPost()['password'])){
 			$this->checkPassProfile();
 		}elseif (isset($_FILES['avatar'])){
 			$this->avatar();
 		}else{
-			echo $this->_twig->render('profile.twig', ['logged' => TRUE,
-			                                          'user'   => $key['pseudo'],
-			                                          'value'  => $key['email'],
-			                                           'server' => $this->_server,
-			                                          'avatar'=> $key['avatar']]);
+			echo $this->getTwig()->render('profile.twig', ['logged' => TRUE,
+			                                               'user'   => $key['pseudo'],
+			                                               'value'  => $key['email'],
+			                                               'admin' => $key['admin'],
+			                                               'server' => $this->getServer(),
+			                                               'avatar'=> $key['avatar']]);
 		}
 	}
 
+	/**
+	 * Affiche les commentaire à valider sur la page admin.
+	 */
+
 	public function admin(){
 
 		$key = (new Session)->getKey('user');
-		$admin = $this->_adminManager->checkAdmin((int)$key['id']);
+		$adminManager = (new AdminManager());
+		$admin = $adminManager->checkAdmin((int)$key['id']);
 
 
 		if ($admin){
-			$allComments = $this->_adminManager->commentsPost();
-			echo $this->_twig->render('admin.twig', ['logged' => TRUE,
-												     'user'   => $key['pseudo'],
-												     'allComments'  => $allComments,
-												     'server' => $this->_server,]);
+			$allComments = $adminManager->commentsPost();
+			echo $this->getTwig()->render('admin.twig', ['logged' => TRUE,
```

Comment:
Codacy found an issue: [Use of echo language construct is discouraged.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=6061935)